storage/remote/queue_manager_test.go

```diff
@@ -2332,24 +2332,147 @@ func BenchmarkBuildTimeSeries(b *testing.B) {
+		},
+		{
+			name:                    "gauge_with_bytes",
+			typeLabel:               "gauge",
+			unitLabel:               "bytes",
+			expectedType:            writev2.Metadata_METRIC_TYPE_GAUGE,
+			description:             "Gauge metric with bytes unit",
+			enableTypeAndUnitLabels: true,
+		},
+		{
+			name:                    "histogram_with_seconds",
+			typeLabel:               "histogram",
+			unitLabel:               "seconds",
+			expectedType:            writev2.Metadata_METRIC_TYPE_HISTOGRAM,
+			description:             "Histogram metric with seconds unit",
+			enableTypeAndUnitLabels: true,
+		},
+		{
+			name:                    "summary_with_ratio",
+			typeLabel:               "summary",
+			unitLabel:               "ratio",
+			expectedType:            writev2.Metadata_METRIC_TYPE_SUMMARY,
+			description:             "Summary metric with ratio unit",
+			enableTypeAndUnitLabels: true,
+		},
+		{
+			name:                    "info_no_unit",
+			typeLabel:               "info",
+			unitLabel:               "",
+			expectedType:            writev2.Metadata_METRIC_TYPE_INFO,
+			description:             "Info metric without unit",
+			enableTypeAndUnitLabels: true,
+		},
+		{
+			name:                    "stateset_no_unit",
+			typeLabel:               "stateset",
+			unitLabel:               "",
+			expectedType:            writev2.Metadata_METRIC_TYPE_STATESET,
+			description:             "Stateset metric without unit",
+			enableTypeAndUnitLabels: true,
+		},
+		{
+			name:                    "unknown_type",
+			typeLabel:               "unknown_type",
+			unitLabel:               "meters",
+			expectedType:            writev2.Metadata_METRIC_TYPE_UNSPECIFIED,
+			description:             "Unknown type defaults to unspecified",
+			enableTypeAndUnitLabels: true,
+		},
+		{
+			name:                    "empty_type_with_unit",
+			typeLabel:               "",
+			unitLabel:               "watts",
+			expectedType:            writev2.Metadata_METRIC_TYPE_UNSPECIFIED,
+			description:             "Empty type with unit",
+			enableTypeAndUnitLabels: true,
+		},
+		{
+			name:                    "type_no_unit",
+			typeLabel:               "gauge",
+			unitLabel:               "",
+			expectedType:            writev2.Metadata_METRIC_TYPE_GAUGE,
+			description:             "Type without unit",
+			enableTypeAndUnitLabels: true,
+		},
+		{
+			name:                    "no_type_no_unit",
+			typeLabel:               "",
+			unitLabel:               "",
+			expectedType:            writev2.Metadata_METRIC_TYPE_UNSPECIFIED,
+			description:             "No type and no unit",
+			enableTypeAndUnitLabels: true,
+		},
 	}
 
-	nSamples, nExemplars, nHistograms, nMetadata, nUnexpected := populateV2TimeSeries(
-		&symbolTable, batch, pendingData, false, false,
-	)
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			batch := make([]timeSeries, 1)
+			builder := labels.NewScratchBuilder(2)
```

Comment:
Again, `schema` has helpers for adding those labels

Reply:
https://github.com/prometheus/prometheus/pull/17033/commits/f8749776988443a81c790c6f10c2fb59aa147a5a